cmd/clusterctl/examples/vsphere/machines.yaml.template

```diff
@@ -4,7 +4,7 @@ items:
   - apiVersion: "cluster.k8s.io/v1alpha1"
     kind: Machine
     metadata:
-      generateName: vs-master-
+      name: ${CLUSTER_NAME}-mgmt
```

Comment:
Hi @andrewsykim,

Is `generateName` not something CAPV uses? Or is it some old CAPI label no longer used?

Reply:
The `generateName` field is confusing IMO because it only works from the context of `clusterctl`. If you try to apply a machine resource with `generateName` it just errors on validation

Reply:
Huh. That *is* odd. Do we expect the templates to be used for anything *but* `clusterctl` though?

Reply:
We don't, BUT I imagine the generated machine objects will be referenced for future machines created against the management cluster

Reply:
Hi @andrewsykim,

Maybe it's the time of day, but I've read this:

> I imagine the generated machine objects will be referenced for future machines created against the management cluster

six times, and I still don't understand what you mean. :)

Reply:
So `generate-yaml.sh` generates a cluster, machine and machinset resource. This is used by clusterctl for the bootstrap cluster that is eventually pivoted into the mgmt cluster. I imagine that the generated YAML will be the first _working_ cluster, machine and machinset spec that most users of CAPV will have and will likely use it (as a reference) for target clusters created against the mgmt cluster created by clusterctl. 

If you try to apply a machine resource with the `generateName` field directly against the mgmt cluster (i.e. not with clusterctl), it will not work s

Reply:
the name `-mgmt` is not clear to me, how about `-controlplane` or `-master`?

Reply:
Not `-master` as that convention is one we actively want to avoid.

Reply:
agree, we should use `-controlplane`

Reply:
From my perspective, appending `-controlplane` to the management cluster nodes is redundant because I don't think we would advise adding non control plane nodes there.

Reply:
could you clarify what do you mean regards management cluster? i could be out of date in term of naming, there are `bootstrap cluster` and `target cluster`, this `machine.yaml` is for controlplace node of the `target cluster`, whether it been used as management cluster node or not all falls under user's preferences, but many of previous discussions are actually around bootstrap cluster functioning as management cluster.

Reply:
Hi @figo,

We're not longer supporting a non-pivot deployment strategy, so the management cluster is the initial target cluster deployed by the bootstrap cluster, to which management resources are pivoted.

Reply:
is there doc or issue about the latest design? i definitely need to understand more about rationale behind it,  @sidharthsurana

Reply:
Remember this is also just an example walk through. The example generated yaml DOES use bootstrap, if you want your own management cluster you decide the name of the node anyways

Reply:
> Remember this is also just an example walk through. The example generated yaml DOES use bootstrap, if you want your own management cluster you decide the name of the node anyways

I agree this is confusing though. As I've mentioned elsewhere, it's super unclear if the purpose of the templates is for the initial target cluster only, or to be used to create additional clusters later.

Reply:
I will try to make this clear in the docs coming up, naming the node to something like `vpshere-mgmt-test1` like I mentioned before might make this super clear. Thoughts?

Reply:
I think a basic PlantUML diagram we can put on our README with names like "bootstrap cluster", "initial target/management cluster", "additional cluster n-n-1" will be useful.

Reply:
Updated the default cluster name to `capv-mgmt-example` and the machine to `capv-mgmt-example-controlplane`